src/components/calcite-accordion/calcite-accordion.tsx

```diff
@@ -0,0 +1,212 @@
+
+  @Element() el: HTMLElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /** specify the theme of dropdrown, defaults to light */
+  @Prop({ mutable: true, reflect: true }) theme: "light" | "dark" = "light";
+
+  /** specify the scale of dropdrown, defaults to m */
+  @Prop({ mutable: true, reflect: true }) scale: "s" | "m" | "l" = "m";
+
+  /** specify the appearance, defaults to default */
+  @Prop({ mutable: true, reflect: true }) appearance: "default" | "minimal" =
+    "default";
+
+  /** specify the placement of the icon, defaults to end */
+  @Prop({ mutable: true, reflect: true }) iconPosition: "start" | "end" = "end";
+
+  /** specify the selection mode - defaults to multi */
+  @Prop({ mutable: true, reflect: true }) selectionMode:
+    | "multi"
+    | "single"
+    | "single-persist" = "multi";
+
+  //--------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+
+  @Event() calciteAccordionItemHasChanged: EventEmitter;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
+
+  connectedCallback() {
+    // validate props
+    let appearance = ["default", "minimal"];
+    if (!appearance.includes(this.appearance)) this.appearance = "default";
+
+    let iconPosition = ["start", "end"];
+    if (!iconPosition.includes(this.iconPosition)) this.iconPosition = "end";
+
+    let theme = ["light", "dark"];
+    if (!theme.includes(this.theme)) this.theme = "light";
+
+    let scale = ["s", "m", "l"];
+    if (!scale.includes(this.scale)) this.scale = "m";
+
+    let selectionMode = ["multi", "single", "single-persist"];
+    if (!selectionMode.includes(this.selectionMode))
+      this.selectionMode = "multi";
+  }
+
+  componentDidLoad() {
+    if (!this.sorted) {
+      this.items = this.sortItems(this.items);
+      this.sorted = true;
+    }
+  }
+
+  render() {
+    const dir = getElementDir(this.el);
+    return (
+      <Host dir={dir} id={this.accordionId} tabindex="-1">
+        <div class="calcite-accordion-wrapper" role="menu">
+          <slot />
+        </div>
+      </Host>
+    );
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+
+  @Listen("calciteAccordionItemKeyEvent") calciteAccordionItemKeyEvent(
+    item: CustomEvent
+  ) {
+    let e = item.detail.item;
+    let itemToFocus = e.target;
+    let isFirstItem = this.itemIndex(itemToFocus) === 0;
+    let isLastItem = this.itemIndex(itemToFocus) === this.items.length - 1;
+    switch (e.keyCode) {
```

Comment:
same as above, switch to event.key

Reply:
Isn't `event.key` slightly different in IE/Edge? I seem to recall some small compatibility issues between the key names on diff browsers.

Reply:
Tested on Edge. Seemed fine.
https://jsfiddle.net/pr3tori4n/xys9avgt/
Codepen and JSfiddle have stopped supporting IE11. Browser needs to die already.

Reply:
Looks like the differences are:

```
"Win" => "Meta"
"Scroll" => "ScrollLock"
"Spacebar" => " "

"Down" => "ArrowDown"
"Left" => "ArrowLeft"
"Right" => "ArrowRight"
"Up" => "ArrowUp"

"Del" => "Delete"
"Apps" => "ContextMenu"
"Esc" => "Escape"

"Multiply" => "*"
"Add" => "+"
"Subtract" => "-"
"Decimal" => "."
"Divide" => "/"
```

Reply:
+ 1 to @paulcpederson, `event.key` is a huge ergonomic improvement but Microsoft won't let us have it.

Reply:
So for our purposes it's really just:

```
"Spacebar" => " "
"Down" => "ArrowDown"
"Left" => "ArrowLeft"
"Right" => "ArrowRight"
"Up" => "ArrowUp"
"Del" => "Delete"
"Esc" => "Escape"
```

as we only really do a11y keyboard navigation stuff

Reply:
I think the best way to handle this would be a `getKey` util like:

```
import {getKey} from '../utils'

function (e: KeyboardEvent) {
  const key = getKey(e.key);
  // do some stuff with normalized key value
}
```

Reply:
A company I used to work at did a lookup based on the keycode, like

```ts
const keys = {
  spacebar: 32,
  leftArrow: 37,
  rightArrow: 39,
  // etc.
}

eventHandler(e: Event) {
  if (e.key === keys.spacebar) {
    // do spacebar stuff
  }
}
```

That gives you some of the benefits of semantic key naming, while still using the cross-browser keycodes.

Reply:
We do have the `keys.ts` util file that has those aliases - larger conversation here since it spans across most components https://github.com/Esri/calcite-components/issues/162